build from master failed

cargo +nightly build --release

```
 Compiling rustc-ap-arena v491.0.0
error[E0599]: no method named `cap` found for type `alloc::raw_vec::RawVec<T>` in the current scope
   --> /Users/chenxiang/.cargo/registry/src/mirrors.ustc.edu.cn-12df342d903acd47/rustc-ap-arena-491.0.0/lib.rs:102:47
    |
102 |                 self.start().add(self.storage.cap())
    |                                               ^^^ private field, not a method

error[E0599]: no method named `cap` found for type `alloc::raw_vec::RawVec<T>` in the current scope
   --> /Users/chenxiang/.cargo/registry/src/mirrors.ustc.edu.cn-12df342d903acd47/rustc-ap-arena-491.0.0/lib.rs:273:55
    |
273 |                     new_capacity = last_chunk.storage.cap();
    |                                                       ^^^ private field, not a method

error[E0599]: no method named `cap` found for type `alloc::raw_vec::RawVec<u8>` in the current scope
   --> /Users/chenxiang/.cargo/registry/src/mirrors.ustc.edu.cn-12df342d903acd47/rustc-ap-arena-491.0.0/lib.rs:408:55
    |
408 |                     new_capacity = last_chunk.storage.cap();
    |                                                       ^^^ private field, not a method

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `rustc-ap-arena`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```
```
 (master)$ rustup --version
rustup 1.17.0 ( )
 (master)$ cargo --version
cargo 1.36.0 (c4fcfb725 2019-05-15)
 (master)$ rustc --version
rustc 1.36.0 (a53f9df32 2019-07-03)
 (master)$ rustup run nightly rustup show
Default host: x86_64-apple-darwin

installed toolchains
--------------------

stable-x86_64-apple-darwin
nightly-x86_64-apple-darwin

active toolchain
----------------

nightly-x86_64-apple-darwin (environment override by RUSTUP_TOOLCHAIN)
rustc 1.38.0-nightly (890881f8f 2019-07-25)
```

After applying that patch (or trying the fix branch from the PR), I'm still getting build errors:
```
$ cargo +nightly build --release
    Updating crates.io index
   Compiling semver-parser v0.7.0
   Compiling libc v0.2.48
   Compiling autocfg v0.1.2
   Compiling arrayvec v0.4.10
   Compiling nodrop v0.1.13
   Compiling cfg-if v0.1.6
   Compiling rand_core v0.4.0
   Compiling scopeguard v0.3.3
   Compiling memoffset v0.2.1
   Compiling lazy_static v1.2.0
   Compiling proc-macro2 v0.4.27
   Compiling void v1.0.2
   Compiling stable_deref_trait v1.1.1
   Compiling rustc-rayon-core v0.2.0
   Compiling unicode-xid v0.1.0
   Compiling byteorder v1.3.1
   Compiling rustc-rayon v0.2.0
   Compiling either v1.5.0
   Compiling indexmap v1.0.2
   Compiling rustc-ap-graphviz v541.0.0
   Compiling memchr v2.1.3
   Compiling serde v1.0.87
   Compiling ryu v0.2.7
   Compiling unicode-width v0.1.5
   Compiling scoped-tls v1.0.0
   Compiling ucd-util v0.1.3
   Compiling regex v1.1.0
   Compiling rustc-ap-rustc_target v541.0.0
   Compiling itoa v0.4.3
   Compiling bitflags v1.0.4
   Compiling annotate-snippets v0.6.1
   Compiling rustc-ap-syntax v541.0.0
   Compiling termcolor v1.0.4
   Compiling quick-error v1.2.2
   Compiling utf8-ranges v1.0.2
   Compiling strsim v0.7.0
   Compiling rustc-ap-rustc_lexer v541.0.0
   Compiling vec_map v0.8.1
   Compiling ansi_term v0.11.0
   Compiling lazycell v1.2.1
   Compiling crossbeam-utils v0.2.2
   Compiling log v0.4.6
   Compiling unreachable v1.0.0
   Compiling crossbeam-utils v0.6.6
   Compiling thread_local v0.3.6
   Compiling owning_ref v0.4.0
   Compiling rand_core v0.3.1
   Compiling itertools v0.8.0
   Compiling rand_chacha v0.1.1
   Compiling rand v0.6.5
   Compiling textwrap v0.10.0
   Compiling semver v0.9.0
   Compiling humantime v1.2.0
   Compiling regex-syntax v0.6.5
   Compiling smallvec v0.6.8
   Compiling ena v0.13.0
   Compiling rand_isaac v0.1.1
   Compiling rand_hc v0.1.0
   Compiling rand_xorshift v0.1.1
   Compiling lock_api v0.1.5
   Compiling rustc-ap-serialize v541.0.0
   Compiling rustc_version v0.2.3
   Compiling crossbeam-epoch v0.3.1
   Compiling rand_os v0.1.2
   Compiling rand_jitter v0.1.3
   Compiling num_cpus v1.9.0
   Compiling jobserver v0.1.13
   Compiling atty v0.2.11
   Compiling rustc-hash v1.0.1
   Compiling clap v2.32.0
   Compiling rand_pcg v0.1.1
   Compiling parking_lot_core v0.4.0
   Compiling crossbeam-deque v0.2.0
   Compiling aho-corasick v0.6.9
   Compiling quote v0.6.11
   Compiling syn v0.15.26
   Compiling parking_lot v0.7.1
   Compiling rustc-ap-rustc_data_structures v541.0.0
   Compiling rustc-ap-arena v541.0.0
   Compiling env_logger v0.6.0
   Compiling synstructure v0.10.2
   Compiling serde_derive v1.0.87
   Compiling derive_more v0.13.0
   Compiling rustc-ap-rustc_macros v541.0.0
   Compiling rustc-ap-syntax_pos v541.0.0
   Compiling rustc-ap-rustc_errors v541.0.0
error: expected one of `,`, `@`, or `]`, found `..`
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:114:28
    |
114 |                 ["a", ref a..] => dl.aggregate_align = align(a, "a")?,
    |                            ^^ expected one of `,`, `@`, or `]` here

error: expected one of `,`, `@`, or `]`, found `..`
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:115:30
    |
115 |                 ["f32", ref a..] => dl.f32_align = align(a, "f32")?,
    |                              ^^ expected one of `,`, `@`, or `]` here

error: expected one of `,`, `@`, or `]`, found `..`
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:116:30
    |
116 |                 ["f64", ref a..] => dl.f64_align = align(a, "f64")?,
    |                              ^^ expected one of `,`, `@`, or `]` here

error: expected one of `,`, `@`, or `]`, found `..`
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:117:35
    |
117 |                 [p @ "p", s, ref a..] | [p @ "p0", s, ref a..] => {
    |                                   ^^ expected one of `,`, `@`, or `]` here

error: expected one of `,`, `@`, or `]`, found `..`
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:117:60
    |
117 |                 [p @ "p", s, ref a..] | [p @ "p0", s, ref a..] => {
    |                                                            ^^ expected one of `,`, `@`, or `]` here

error: expected one of `,`, `@`, or `]`, found `..`
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:121:26
    |
121 |                 [s, ref a..] if s.starts_with("i") => {
    |                          ^^ expected one of `,`, `@`, or `]` here

error: expected one of `,`, `@`, or `]`, found `..`
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:145:26
    |
145 |                 [s, ref a..] if s.starts_with("v") => {
    |                          ^^ expected one of `,`, `@`, or `]` here

error[E0308]: mismatched types
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:114:62
    |
114 |                 ["a", ref a..] => dl.aggregate_align = align(a, "a")?,
    |                                                              ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:115:58
    |
115 |                 ["f32", ref a..] => dl.f32_align = align(a, "f32")?,
    |                                                          ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:116:58
    |
116 |                 ["f64", ref a..] => dl.f64_align = align(a, "f64")?,
    |                                                          ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:119:46
    |
119 |                     dl.pointer_align = align(a, p)?;
    |                                              ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:129:35
    |
129 |                     let a = align(a, s)?;
    |                                   ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error[E0308]: mismatched types
   --> /Users/david/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_target-541.0.0/abi/mod.rs:147:35
    |
147 |                     let a = align(a, s)?;
    |                                   ^ expected slice, found &str
    |
    = note: expected type `&[&str]`
               found type `&&str`

error: aborting due to 13 previous errors

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `rustc-ap-rustc_target`.
warning: build failed, waiting for other jobs to finish...
error: build failed
Davids-MacBook-Pro-2 ~/projects/racer
$
```

与楼主一样的问题。
四天了还没解决，进度有点慢啊。

error[E0599]: no method named `cap` found for type `alloc::raw_vec::RawVec<T>` in the current scope
   --> C:\Users\Administrator\.cargo\registry\src\github.com-1ecc6299db9ec823\rustc-ap-arena-491.0.0\lib.rs:102:47
    |
102 |                 self.start().add(self.storage.cap())
    |                                               ^^^ private field, not a method

error[E0599]: no method named `cap` found for type `alloc::raw_vec::RawVec<T>` in the current scope
   --> C:\Users\Administrator\.cargo\registry\src\github.com-1ecc6299db9ec823\rustc-ap-arena-491.0.0\lib.rs:273:55
    |
273 |                     new_capacity = last_chunk.storage.cap();
    |                                                       ^^^ private field, not a method

error[E0599]: no method named `cap` found for type `alloc::raw_vec::RawVec<u8>` in the current scope
   --> C:\Users\Administrator\.cargo\registry\src\github.com-1ecc6299db9ec823\rustc-ap-arena-491.0.0\lib.rs:408:55
    |
408 |                     new_capacity = last_chunk.storage.cap();
    |                                                       ^^^ private field, not a method

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `rustc-ap-arena`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `racer v2.1.23`, intermediate artifacts can be found at `C:\Users\ADMINI~1\AppData\Local\Temp\cargo-installp53wWA`

Caused by:
  build failed

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@ debug = false # because of #1005
 [dependencies]
 bitflags = "1.0"
 log = "0.4"
-rustc-ap-syntax = "491.0.0"
+rustc-ap-syntax = "541.0.0"
 env_logger = "0.6"
 clap = "2.32"
 lazy_static = "1.2"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@ debug = false # because of #1005
 [dependencies]
 bitflags = "1.0"
 log = "0.4"
-rustc-ap-syntax = "541.0.0"
+rustc-ap-syntax = "542.0.0"
 env_logger = "0.6"
 clap = "2.32"
 lazy_static = "1.2"

--- a/src/racer/ast.rs
+++ b/src/racer/ast.rs
@@ -272,58 +272,26 @@ fn destructure_pattern_to_ty(
         }
         PatKind::Tuple(ref tuple_elements) => match *ty {
             Ty::Tuple(ref typeelems) => {
-                let mut res = None;
                 for (i, p) in tuple_elements.iter().enumerate() {
                     if !point_is_in_span(point, &p.span) {
                         continue;
                     }
                     if let Some(ref ty) = typeelems[i] {
-                        res = destructure_pattern_to_ty(p, point, ty, scope, session);
-                        break;
+                        return destructure_pattern_to_ty(p, point, ty, scope, session);
                     }
                 }
-                res
+                None
             }
             _ => panic!("Expecting TyTuple"),
         },
         PatKind::TupleStruct(ref path, ref children) => {
-            let m = resolve_ast_path(path, &scope.filepath, scope.point, session);
+            let m = resolve_ast_path(path, &scope.filepath, scope.point, session)?;
             let contextty = path_to_match(ty.clone(), session);
-            if let Some(m) = m {
-                let mut res = None;
-                for (i, p) in children.iter().enumerate() {
-                    if point_is_in_span(point, &p.span) {
-                        res = typeinf::get_tuplestruct_field_type(i, &m, session)
-                            .and_then(|ty|
-                                // if context ty is a match, use its generics
-                                if let Some(Ty::Match(ref contextm)) = contextty {
-                                    path_to_match_including_generics(ty, contextm.to_generics(), session)
-                                } else {
-                                    path_to_match(ty, session)
-                                })
-                            .and_then(|ty| destructure_pattern_to_ty(p, point, &ty, scope, session));
-                        break;
-                    }
-                }
-                res
-            } else {
-                None
-            }
-        }
-        PatKind::Struct(ref path, ref children, _) => {
-            let m = resolve_ast_path(path, &scope.filepath, scope.point, session);
-            let contextty = path_to_match(ty.clone(), session);
-            if let Some(m) = m {
-                let mut res = None;
-
-                for child in children {
-                    if point_is_in_span(point, &child.span) {
-                        res = typeinf::get_struct_field_type(
-                            &child.node.ident.name.as_str(),
-                            &m,
-                            session,
-                        )
+            for (i, p) in children.iter().enumerate() {
+                if point_is_in_span(point, &p.span) {
+                    return typeinf::get_tuplestruct_field_type(i, &m, session)
                         .and_then(|ty| {
+                            // if context ty is a match, use its generics
                             if let Some(Ty::Match(ref contextm)) = contextty {
                                 path_to_match_including_generics(
                                     ty,
@@ -334,18 +302,34 @@ fn destructure_pattern_to_ty(
                                 path_to_match(ty, session)
                             }
                         })
-                        .and_then(|ty| {
-                            destructure_pattern_to_ty(&child.node.pat, point, &ty, scope, session)
-                        });
-
-                        break;
-                    }
+                        .and_then(|ty| destructure_pattern_to_ty(p, point, &ty, scope, session));
                 }
-
-                res
-            } else {
-                None
             }
+            None
+        }
+        PatKind::Struct(ref path, ref children, _) => {
+            let m = resolve_ast_path(path, &scope.filepath, scope.point, session)?;
+            let contextty = path_to_match(ty.clone(), session);
+            for child in children {
+                if point_is_in_span(point, &child.span) {
+                    return typeinf::get_struct_field_type(
+                        &child.node.ident.name.as_str(),
+                        &m,
+                        session,
+                    )
+                    .and_then(|ty| {
+                        if let Some(Ty::Match(ref contextm)) = contextty {
+                            path_to_match_including_generics(ty, contextm.to_generics(), session)
+                        } else {
+                            path_to_match(ty, session)
+                        }
+                    })
+                    .and_then(|ty| {
+                        destructure_pattern_to_ty(&child.node.pat, point, &ty, scope, session)
+                    });
+                }
+            }
+            None
         }
         _ => {
             debug!("Could not destructure pattern {:?}", pat);
@@ -1329,7 +1313,8 @@ impl<'ast, 'r, 's> visit::Visitor<'ast> for IfLetVisitor<'r, 's> {
                 if let ExprKind::Let(pats, expr) = &let_stmt.node {
                     if let Some(pat) = pats.get(0) {
                         self.let_pat = Some(Pat::from_ast(&pat.node, &self.scope));
-                        let mut expr_visitor = ExprTypeVisitor::new(self.scope.clone(), self.session);
+                        let mut expr_visitor =
+                            ExprTypeVisitor::new(self.scope.clone(), self.session);
                         expr_visitor.visit_expr(expr);
                         self.rh_expr = expr_visitor.result;
                     }

--- a/src/racer/ast.rs
+++ b/src/racer/ast.rs
@@ -7,7 +7,7 @@ use typeinf;
 use std::path::Path;
 use std::rc::Rc;
 
-use syntax::ast::{self, ExprKind, FunctionRetTy, ItemKind, PatKind, UseTree, UseTreeKind};
+use syntax::ast::{self, Expr, ExprKind, FunctionRetTy, ItemKind, PatKind, UseTree, UseTreeKind};
 use syntax::edition::Edition;
 use syntax::errors::{emitter::Emitter, DiagnosticBuilder, Handler};
 use syntax::parse::parser::Parser;
@@ -165,11 +165,13 @@ impl<'ast> visit::Visitor<'ast> for PatBindVisitor {
 
     fn visit_expr(&mut self, ex: &ast::Expr) {
         // don't visit the RHS or block of an 'if let' or 'for' stmt
-        match ex.node {
-            ExprKind::IfLet(ref pat, ..) | ExprKind::WhileLet(ref pat, ..) => {
-                pat.iter().for_each(|pat| self.visit_pat(pat))
+        match &ex.node {
+            ExprKind::If(let_stmt, ..) | ExprKind::While(let_stmt, ..) => {
+                if let ExprKind::Let(pats, ..) = &let_stmt.node {
+                    pats.iter().for_each(|pat| self.visit_pat(pat))
+                }
             }
-            ExprKind::ForLoop(ref pat, ..) => self.visit_pat(pat),
+            ExprKind::ForLoop(pat, ..) => self.visit_pat(pat),
             _ => visit::walk_expr(self, ex),
         }
     }
@@ -715,8 +717,7 @@ impl<'c, 's, 'ast> visit::Visitor<'ast> for ExprTypeVisitor<'c, 's> {
                     }
                 }
             }
-            ExprKind::If(_, ref block, ref else_block)
-            | ExprKind::IfLet(_, _, ref block, ref else_block) => {
+            ExprKind::If(_, ref block, ref else_block) => {
                 debug!("if/iflet expr");
                 if let Some(stmt) = block.stmts.last() {
                     visit::walk_stmt(self, stmt);
@@ -1326,12 +1327,14 @@ pub(crate) struct IfLetVisitor<'r, 's: 'r> {
 impl<'ast, 'r, 's> visit::Visitor<'ast> for IfLetVisitor<'r, 's> {
     fn visit_expr(&mut self, ex: &'ast ast::Expr) {
         match &ex.node {
-            ExprKind::IfLet(pats, expr, _, _) | ExprKind::WhileLet(pats, expr, _, _) => {
-                if let Some(pat) = pats.get(0) {
-                    self.let_pat = Some(Pat::from_ast(&pat.node, &self.scope));
-                    let mut expr_visitor = ExprTypeVisitor::new(self.scope.clone(), self.session);
-                    expr_visitor.visit_expr(expr);
-                    self.rh_expr = expr_visitor.result;
+            ExprKind::If(let_stmt, ..) => {
+                if let ExprKind::Let(pats, expr) = &let_stmt.node {
+                    if let Some(pat) = pats.get(0) {
+                        self.let_pat = Some(Pat::from_ast(&pat.node, &self.scope));
+                        let mut expr_visitor = ExprTypeVisitor::new(self.scope.clone(), self.session);
+                        expr_visitor.visit_expr(expr);
+                        self.rh_expr = expr_visitor.result;
+                    }
                 }
             }
             _ => {}

--- a/src/racer/ast.rs
+++ b/src/racer/ast.rs
@@ -7,7 +7,7 @@ use typeinf;
 use std::path::Path;
 use std::rc::Rc;
 
-use syntax::ast::{self, Expr, ExprKind, FunctionRetTy, ItemKind, PatKind, UseTree, UseTreeKind};
+use syntax::ast::{self, ExprKind, FunctionRetTy, ItemKind, PatKind, UseTree, UseTreeKind};
 use syntax::edition::Edition;
 use syntax::errors::{emitter::Emitter, DiagnosticBuilder, Handler};
 use syntax::parse::parser::Parser;
@@ -270,7 +270,7 @@ fn destructure_pattern_to_ty(
                 );
             }
         }
-        PatKind::Tuple(ref tuple_elements, _) => match *ty {
+        PatKind::Tuple(ref tuple_elements) => match *ty {
             Ty::Tuple(ref typeelems) => {
                 let mut res = None;
                 for (i, p) in tuple_elements.iter().enumerate() {
@@ -286,12 +286,11 @@ fn destructure_pattern_to_ty(
             }
             _ => panic!("Expecting TyTuple"),
         },
-        PatKind::TupleStruct(ref path, ref children, _) => {
+        PatKind::TupleStruct(ref path, ref children) => {
             let m = resolve_ast_path(path, &scope.filepath, scope.point, session);
             let contextty = path_to_match(ty.clone(), session);
             if let Some(m) = m {
                 let mut res = None;
-
                 for (i, p) in children.iter().enumerate() {
                     if point_is_in_span(point, &p.span) {
                         res = typeinf::get_tuplestruct_field_type(i, &m, session)
@@ -303,7 +302,6 @@ fn destructure_pattern_to_ty(
                                     path_to_match(ty, session)
                                 })
                             .and_then(|ty| destructure_pattern_to_ty(p, point, &ty, scope, session));
-
                         break;
                     }
                 }
@@ -1327,7 +1325,7 @@ pub(crate) struct IfLetVisitor<'r, 's: 'r> {
 impl<'ast, 'r, 's> visit::Visitor<'ast> for IfLetVisitor<'r, 's> {
     fn visit_expr(&mut self, ex: &'ast ast::Expr) {
         match &ex.node {
-            ExprKind::If(let_stmt, ..) => {
+            ExprKind::If(let_stmt, ..) | ExprKind::While(let_stmt, ..) => {
                 if let ExprKind::Let(pats, expr) = &let_stmt.node {
                     if let Some(pat) = pats.get(0) {
                         self.let_pat = Some(Pat::from_ast(&pat.node, &self.scope));

--- a/src/racer/ast_types.rs
+++ b/src/racer/ast_types.rs
@@ -268,6 +268,7 @@ pub enum Pat {
     Range,
     Slice,
     Mac,
+    Rest,
 }
 
 impl Pat {
@@ -305,7 +306,7 @@ impl Pat {
                     .collect();
                 Pat::Struct(path, fields)
             }
-            PatKind::TupleStruct(path, pats, _) => {
+            PatKind::TupleStruct(path, pats) => {
                 let path = Path::from_ast(path, scope);
                 let pats = pats
                     .iter()
@@ -314,7 +315,7 @@ impl Pat {
                 Pat::TupleStruct(path, pats)
             }
             PatKind::Path(_, path) => Pat::Path(Path::from_ast(&path, scope)),
-            PatKind::Tuple(pats, _) => {
+            PatKind::Tuple(pats) => {
                 let pats = pats
                     .iter()
                     .map(|pat| Pat::from_ast(&pat.node, scope))
@@ -329,6 +330,7 @@ impl Pat {
             // ignore paren
             PatKind::Paren(pat) => Pat::from_ast(&pat.node, scope),
             PatKind::Mac(_) => Pat::Mac,
+            PatKind::Rest => Pat::Rest,
         }
     }
 }

--- a/src/racer/core.rs
+++ b/src/racer/core.rs
@@ -1246,27 +1246,24 @@ where
 /// let session = racer::Session::new(&cache, None);
 ///
 /// // This is the file where we request completion from
-/// let src = stringify! {
+/// let src = r"
 ///    mod sub;
 ///    use sub::foo;
 ///    fn main() {
 ///        foo();
 ///    };
-/// };
+/// ";
 ///
 /// // This is the submodule where the definition is found
-/// let sub = stringify! {
-///     pub fn foo() {}
-/// };
+/// let sub = r"pub fn foo() {}";
 ///
 /// // Load files into cache to prevent trying to read from disk
 /// session.cache_file_contents("sub.rs", sub);
 /// session.cache_file_contents("lib.rs", src);
 ///
-/// // Search for the definition. 45 is the byte offset
-/// // in `src` after stringify! runs. Specifically, this asks
-/// // for the definition of `foo()`.
-/// let m = racer::find_definition("lib.rs", racer::Location::from(45), &session)
+/// // Search for the definition. 52 is the byte offset in `src`.
+/// // Specifically, this asks for the definition of `foo()`.
+/// let m = racer::find_definition("lib.rs", racer::Location::from(52), &session)
 ///               .expect("find definition returns a match");
 ///
 /// // Should have found definition in the "sub.rs" file

--- a/tests/system.rs
+++ b/tests/system.rs
@@ -3580,6 +3580,8 @@ fn doesnt_complete_macro_after_use() {
 }
 
 #[test]
+#[ignore]
+// FIXME:  #1059
 fn complets_stringify() {
     let src = r#"
     fn main() {
@@ -3614,6 +3616,8 @@ fn completes_vec() {
 }
 
 #[test]
+#[ignore]
+// FIXME: #1059
 fn finds_std_macro_doc() {
     let src = r#"
     fn main() {

